templates/tweets/scripts.html

```diff
@@ -0,0 +1,56 @@
+<script type="text/javascript">
+  const getCookie = name => {
+    if (document.cookie && document.cookie !== "") {
+      for (const cookie of document.cookie.split(";")) {
+        const [key, value] = cookie.trim().split("=");
+        if (key === name) {
+          return decodeURIComponent(value);
+        }
+      }
+    }
+  };
+  const csrftoken = getCookie('csrftoken');
+
+
+  const changeStyles = (jsonResponse,func) => {
+    const selector = document.getElementsByName(jsonResponse.tweet_id)[0];
+    if (jsonResponse.liked) {
+      console.log("liked")
+      selector.outerHTML = "<button data-button=\"like\" data-url=" + "\"{% url 'tweets:unlike' 0 %}\"".replace("0", jsonResponse.tweet_id) + "name=\"{{tweet.id}}\"><i class=\"fas fa-lg fa-heart like-red\"></i></button>";
+    } else {
+      console.log("not liked")
+      selector.outerHTML = "<button data-button=\"like\" data-url=" + "\"{% url 'tweets:like' 0 %}\"".replace("0", jsonResponse.tweet_id) + " name=\"{{tweet.id}}\"><i class=\"far fa-lg fa-heart\"></i></button>";
+    }
+    document.getElementsByName(jsonResponse.tweet_id + "-count")[0].textContent = jsonResponse.count;
+    // イベントリスナーの再設定
+    document.getElementsByName(jsonResponse.tweet_id)[0].addEventListener("click", func);
+  }
```

Comment:
```suggestion
  const changeStyles = (jsonResponse) => {
    const selector = document.getElementsByName(jsonResponse.tweet_id)[0];
    const url = selector.getAttribute('data-url');
    const split_url = url.split('/');
    if (jsonResponse.liked) {
      console.log("liked")
      unlike = url.replace(split_url[3], 'unlike');
      selector.setAttribute('data-url', unlike);
      selector.innerHTML = "<i class='fas fa-lg fa-heart like-red'></i>";
    } else {
      console.log("not liked")
      like = url.replace(split_url[3], 'like');
      selector.setAttribute('data-url', like);
      selector.innerHTML = "<i class='far fa-lg fa-heart'></i>";
    }
    document.getElementsByName(jsonResponse.tweet_id + "-count")[0].textContent = jsonResponse.count;
  }
```
data-urlは以上のような書き方をして書き換えてみました。このようにした理由は、なるべくハードコーディングを避けるためです。とはいえ、like, unlike書いてしまっているので、ハードコーディング気味ではありますね。
もちろん、今回22行目で書いてもらったように直接書いてしまっても大丈夫です。

Reply:
修正しました。